src/ocp.erl

```diff
@@ -60,19 +95,30 @@ finish_span() ->
 
 %%--------------------------------------------------------------------
 %% @doc
-%% Starts a new span as a child of the current span and pushes the
-%% parent to the open span stack in the process dictionary.
+%% Return the current trace context for the process.
 %% @end
 %%--------------------------------------------------------------------
--spec child_span(unicode:chardata()) -> opencensus:maybe(opencensus:span()).
-child_span(Name) ->
+-spec context() -> opencensus:maybe(opencensus:span()).
+context() ->
     Span = get(?KEY),
-    NewSpan = opencensus:child_span(Name, Span),
-    put(?KEY, NewSpan),
-    case get(?SKEY) of
-        undefined ->
-            put(?SKEY, [Span]);
-        Spans ->
-            put(?SKEY, [Span | Spans])
-    end,
-    NewSpan.
+    opencensus:context(Span).
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Put an attribute (a key/value pair) in the attribute map of a span.
+%% If the attribute already exists it is overwritten with the new value.
+%% @end
+%%--------------------------------------------------------------------
+-spec put_attribute(unicode:chardata(), opencensus:attribute_value()) -> ok.
+put_attribute(Key, Value) ->
```

Comment:
guards to protect against bad key values might be nice (i.e. I could see someone trying to use atom keys and then everything exploding)